1-Draft/RFCNNNN-Native-Support-for-Interface-Declarations.md

```diff
@@ -0,0 +1,102 @@
+---
+RFC: RFCNNNN
+Author: Mathias R. Jessen
+Status: Draft
+SupercededBy: N/A
+Version: 0.1
+Area: Language
+Comments Due: 2019-01-04
+Plan to implement: Yes
+---
+
+# Native Support for .NET Interface Declarations
+
+The PowerShell Classes feature introduced in PowerShell 5.0 allows for type definitions at runtime. Since open sourcing PowerShell Core, one of the most oft-repeated feature request I've heard from community members is the ability to also declare and define interfaces at runtime, usually for ease of implementing common software design pattern such as the Builder pattern.
+
+## Motivation
+
+    As a module author,
+    I can leverage common OO design patterns that require abstract type definitions,
+    so that my implementations are easily communicated and explained to other developers and module authors.
+
+## Specification
+
+Implementing native support for interface declarations require both syntactical and semantic changes to the language engine.
+
+### Syntax
+
+Based on the [original issue](https://github.com/PowerShell/PowerShell/issues/2223) raised by @vors, I imagine a simplified subset of the class syntax, as follows:
+
+```powershell
+# interface declaration with no members
+interface IMyInterface { }

```

Comment:
@JamesWTruher and I agree that breaking is undesirable, and we did find [this one instance](https://github.com/ticketmaster/poshspec/blob/master/Public/Interface.ps1) of a real-world module that would break, but we may still decide to do this via an experimental flag and message heavily around the breakage if we decide the value is high enough